bubblegum/program/README.md

```diff
@@ -1,4 +1,413 @@
+
+Beyond verifying creators at the time of mint, there are `verify_creator` and `unverify_creator` instructions that can be used on existing Compressed NFTs.  Of course these transactions must be signed by the creator's wallet.
+
+### Collection verification
+
+Note that there are no such thing as Compressed Verified Collections.  Collections are still NFTs created in the realm of token-metadata Metadata and Master Edition accounts.  Also note that a collection cannot be set to verified at the time of minting.  Instead, there are instructions to `verify_collection` and `unverify_collection`, as well as a `set_and_verify_collection` for the case where the collection was set during time of mint.  All of these require either the true Collection Authority to be a a signer, or a delegated Collection Authority to be a signer along with providing a Collection Authority Record PDA.  See the Metaplex documentation on [`Certified Collections`](https://docs.metaplex.com/programs/token-metadata/certified-collections) for more information on verifying collections.
+
+### Transfer ownership of an NFT 
+
+Transferring an NFT to a different owner is done using the `transfer` instruction.  Transfers must be signed by either the NFT owner or NFT delegate.  When NFTs are transferred there is no delegate authority.  
+
+### Delegate authority for an NFT
+
+Delegation is done using the `delegate` instruction and the transaction must be signed by the NFT owner.
+
+### Burn an NFT
+
+Compressed NFTs can be permanently destroyed using the `burn` instruction.  Burns must be signed by either the NFT owner or NFT delegate.
+
+### Redeem an NFT and decompress it into an uncompressed Metaplex NFT
+
+Redeeming an NFT is done using the `redeem` instruction.  This instruction removes the leaf from the Merkle tree and creates a voucher PDA account.  The voucher account can be sent to the `decompress_v1` instruction to decompress the NFT into an uncompressed Metaplex NFT.  As mentioned above this will cost rent for the token-metadata Metadata and Master Edition accounts that are created during the decompression process.  Note that after a compressed NFT is redeemed but before it is decompressed, the process can be reversed using `cancel_redeem`.  This puts the compressed NFT back into the Merkle tree.  Redeeming, cancelling redeem, and decompressing all must be signed by the NFT owner.
+
+
+
+TODO MERKLE PROOFS MUST BE SUPPLIED TO SOME FUNCTIONS.
+
+
+## Instructions
+
+### 📄 `create_tree`
+
+This instruction creates a Merkle Tree.
+
+<details>
+  <summary>Accounts</summary>
+
+| Name                  | Writable | Signer | Description
+| --------------------- | :------: | :----: | --
+| `tree_authority`      |    ✅    |        | The [`TreeConfig`](https://github.com/metaplex-foundation/metaplex-program-library/blob/master/bubblegum/program/src/state/mod.rs#L17) PDA account that is initialized by this instruction.
```

Comment:
I think we should use links pointing to blobs from commits vs branches (like `master` in this case), because as the branch changes, the specified line may no longer contain what it did when the link was originally added.

Reply:
We'll be pointing at the same thing we were when the text was written. When that becomes old code, the docs probably need an update or bump as well. Anyway, just mentioning this as a practice I've used/seen before.

Reply:
I guess what I don't like about using a specific commit is that it prioritizes keeping the document good at the expense of referencing outdated code.  As a dev I think I would rather have the link go to the latest code, even if it results in going to the wrong line (for example) because some fields were added to the struct.  Or as another example, if the struct is no longer there I could tell the documentation was out of date, rather than continue reading about the old code.